### PR TITLE
LOOP-1833: Adds a "stale pump data" warning if pump data is too stale for bolus recommendation

### PR DESCRIPTION
**Pull request by rickpasetto**
Note: copy in the warning needs review.

Also, changes the behavior when there is any error in calculating the recommended bolus: instead of defaulting to `0` it shows `--`.

https://tidepool.atlassian.net/browse/LOOP-1833

### CODE REVIEW

**Review by nhamming — APPROVED** [review]
LGTM

**Review by ps2 — APPROVED** [review]
LGTM. This should go through design review, for sure.

**Comment by rickpasetto** [comment]
@ps2 @nhamming I'd love for you guys to take another gander at this.  This attempts to `assertCurrentPumpData()` upon entry to the screen.  I still am trying to figure out the right way to add an additional "spinner" next to the entry field; I'd like that to be in a separate PR, since IMO it is an enhancement, but not required.

File: Loop/View Models/BolusEntryViewModel.swift [hunk 1]
```diff
@@ -111,6 +118,12 @@ final class BolusEntryViewModel: ObservableObject {
             .receive(on: DispatchQueue.main)
             .sink { [weak self] _ in self?.update() }
             .store(in: &cancellables)
+        
+        // Look for Loop starting, and assume that if it started after assertCurrentPumpData that we want to call pumpManagerRefreshCompletion
+        NotificationCenter.default
+            .publisher(for: .LoopRunning, object: dataManager.loopManager)
```
Comment (ps2) [inline Loop/View Models/BolusEntryViewModel.swift:124]: Isn't the observeLoopUpdates enough to detect the arrival of fresh pump data?

Comment (rickpasetto) [reply Loop/View Models/BolusEntryViewModel.swift:124]: I thought it did, but not quite.  The above listener (sink) watches for `.LoopDataUpdated` which is called a lot more times than when the pump manage signals it has been refreshed (`pumpManagerRecommendsLoop`).  I wasn't thrilled with all of those updates.

**Pull request merged**